### PR TITLE
conditional update by name

### PR DESCRIPTION
**Pull request by sashrika**
https://issues.openmrs.org/browse/FM-94

### CODE REVIEW

File: omod/src/main/java/org/openmrs/module/fhir/providers/RestfulPatientResourceProvider.java [hunk 1]
```diff
@@ -206,33 +206,45 @@ public MethodOutcome updatePatient(@ResourceParam Patient patient, @IdParam IdDt
...
+			String args[] = theConditional.split("?");
+			String parameterPart = args[1];
+			String paraArgs[] = parameterPart.split("=");
+			String parameterName = paraArgs[0];
+			if ("name".equals(parameterName)) {
+				StringParam param = new StringParam();
+				param.setValue(paraArgs[1]);
+				patientList = patientResource.searchByName(param);
+			} else if ("identifier".equals(parameterName)) {
+				TokenParam params = new TokenParam();
```
Comment (sashrika) [inline omod/src/main/java/org/openmrs/module/fhir/providers/RestfulPatientResourceProvider.java]: Patient?identifier=7C00001 

URL pattern

File: omod/src/main/java/org/openmrs/module/fhir/providers/RestfulPatientResourceProvider.java [hunk 2]
```diff
@@ -206,33 +206,45 @@ public MethodOutcome updatePatient(@ResourceParam Patient patient, @IdParam IdDt
...
+				params.setValue(paraArgs[1]);
+				patientList = patientResource.searchByIdentifier(params);
+			} else if ("givenName".equals(parameterName)) {
+				StringParam param = new StringParam();
```
Comment (sashrika) [inline omod/src/main/java/org/openmrs/module/fhir/providers/RestfulPatientResourceProvider.java]: Patient?name=xyz

File: omod/src/main/java/org/openmrs/module/fhir/providers/RestfulPatientResourceProvider.java [hunk 3]
```diff
@@ -206,33 +206,45 @@ public MethodOutcome updatePatient(@ResourceParam Patient patient, @IdParam IdDt
...
-			name = theConditional.substring(startIndex + 1);
-			TokenParam params = new TokenParam();
-			params.setValue(name);
-			List<Patient> patientList = patientResource.searchByIdentifier(params);
+			List<Patient> patientList = null;
+			String args[] = theConditional.split("?");
+			String parameterPart = args[1];
+			String paraArgs[] = parameterPart.split("=");
+			String parameterName = paraArgs[0];
+			if ("name".equals(parameterName)) {
```
Comment (sashrika) [inline omod/src/main/java/org/openmrs/module/fhir/providers/RestfulPatientResourceProvider.java]: Patient?name=xyz

Comment (harsha89) [inline omod/src/main/java/org/openmrs/module/fhir/providers/RestfulPatientResourceProvider.java]: move name to a constant filed

File: omod/src/main/java/org/openmrs/module/fhir/providers/RestfulPatientResourceProvider.java [hunk 4]
```diff
@@ -206,33 +206,45 @@ public MethodOutcome updatePatient(@ResourceParam Patient patient, @IdParam IdDt
...
-		String name = null;
+		MethodOutcome outcome = new MethodOutcome();
 		if (theConditional != null) {
-			int startIndex = theConditional.lastIndexOf('=');
-			name = theConditional.substring(startIndex + 1);
-			TokenParam params = new TokenParam();
-			params.setValue(name);
-			List<Patient> patientList = patientResource.searchByIdentifier(params);
+			List<Patient> patientList = null;
+			String args[] = theConditional.split("?");
```
Comment (harsha89) [inline omod/src/main/java/org/openmrs/module/fhir/providers/RestfulPatientResourceProvider.java]: possible null pointer if the conditional is null

File: omod/src/main/java/org/openmrs/module/fhir/providers/RestfulPatientResourceProvider.java [hunk 5]
```diff
@@ -206,33 +206,45 @@ public MethodOutcome updatePatient(@ResourceParam Patient patient, @IdParam IdDt
...
+			String parameterPart = args[1];
+			String paraArgs[] = parameterPart.split("=");
+			String parameterName = paraArgs[0];
+			if ("name".equals(parameterName)) {
+				StringParam param = new StringParam();
+				param.setValue(paraArgs[1]);
+				patientList = patientResource.searchByName(param);
+			} else if ("identifier".equals(parameterName)) {
```
Comment (harsha89) [inline omod/src/main/java/org/openmrs/module/fhir/providers/RestfulPatientResourceProvider.java]: move  to a constant filed

File: omod/src/main/java/org/openmrs/module/fhir/providers/RestfulPatientResourceProvider.java (hunk 2 above)
Comment (harsha89) [inline omod/src/main/java/org/openmrs/module/fhir/providers/RestfulPatientResourceProvider.java]: move name to a constant filed

**Comment by sashrika** [comment]
@harsha89  @surangak please review the new changes.

**Pull request merged**